src/server/service/kv.rs

```diff
@@ -1265,8 +1265,22 @@ fn extract_key_error(err: &storage::Error) -> KeyError {
             lock_info.set_lock_ttl(ttl);
             key_error.set_locked(lock_info);
         }
-        storage::Error::Txn(TxnError::Mvcc(MvccError::WriteConflict { .. }))
-        | storage::Error::Txn(TxnError::Mvcc(MvccError::TxnLockNotFound { .. })) => {
+        // failed in prewrite
+        storage::Error::Txn(TxnError::Mvcc(MvccError::WriteConflict {
+            start_ts,
+            conflict_ts,
+            ref key,
+            ref primary,
+        })) => {
+            let mut write_conflict = WriteConflict::new();
+            write_conflict.set_start_ts(start_ts);
+            write_conflict.set_conflict_ts(conflict_ts);
+            write_conflict.set_key(key.to_owned());
```

Comment:
AFAIK, `key` is encoded while `primary` is not,  in the old error string ?
And I expect both of them to be not encoded. @AndreMouche  @disksing

Reply:
I have no idea. @AndreMouche please ease his concern.

Reply:
comment addressed @disksing @tiancaiamao